app.js

```diff
@@ -165,75 +61,219 @@ settings.setPath('app/config/config.json')
 //   })
 // })
 
-ipc.on('close-quick-start', (event, arg) => {
-  quickStartWindow.close()
-})
+function createDocumentWindow () {
+  log.info('Creating document window')
+  global.settings = settings
 
-ipc.on('openQuickStartDialog', (event, arg) => {
-  openQuickStartDialog()
-})
+  // Create the browser window
+  let documentWindow = new BrowserWindow({
+    width: 1350,
+    height: 800,
+    center: true,
+    'min-width': 500,
+    'min-height': 200,
+    'accept-first-mouse': true,
+    'title-bar-style': 'hidden'
+  })
 
-function addToRecent (data) {
-  var recent = settings.get('recent')
-  if (recent.length <= 3)
-    recent.push(data)
-  else
-    recent[0] = data
+  documentWindow.webContents.openDevTools()
 
-  return _.sortBy(recent, 'ts')
+  // and load the index.html of the app.
+  documentWindow.loadURL(
+    url.format({
+      pathname: path.join(__dirname + /app/, 'index.html'),
+      protocol: 'file:',
+      slashes: true
+    })
+  )
+
+  // Emitted when the window is closed.
+  documentWindow.on('closed', function () {
+    log.info('Closing document window')
+    // Dereference the window object, usually you would store windows
+    // in an array if your app supports multi windows, this is the time
+    // when you should delete the corresponding element.
+    documentWindow = null
+  })
+  
+  return documentWindow;
 }
 
-function callAPI (params, cb) {
-  console.log(params)
-  var requestId = guid.raw()
-  log.info(`[${requestId}] ${params.from} fetching ${params.path} from API `)
-
-  var options = {
-    uri: `${params.host}/${params.path}`,
-    headers: {
-      'User-Agent': 'Request-Promise'
-    },
-    json: true,
-    requestId: requestId,
-    from: params.from,
-    host: params.host,
-    path: params.path
+class PokeyConfigurationEditorController {
```

Comment:
Can we base class this ? Seems like there is common functionality across all windows